### PR TITLE
Use consistent type for vreplication id

### PR DESCRIPTION
**Pull request by dbussink**
The `_vt.vreplication` table has an `INT` type without `UNSIGNED`, so it maps to the Go `int32` type to have the exact same consistent range.

This means that we should use `int32` consistently for everywhere that we use the VReplication identifier.

We had int32, uint32 and int used pretty randomly for these which leads to CodeQL flagging these and how this was found.

Instead this now converts everything to int32. It's also safe to change the protobuf, since the protobuf wire format for int32, uint32, int64 & uint64 is all compatible and actually uses the same variable int length format.

The only exception would be if someone has a VReplication identifier that would not fit in `int32`, but well, this PR shows that in that case we already blow up in unexpected and unknown ways anyway. So it's better to have that explicitly error.

## Related Issue(s)

Part of https://github.com/vitessio/vitess/issues/12216

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by mattlord** [comment]
Thank you for these! I had discussed this before with @rohit-nayak-ps but we didn't have Int32 type support in the sqltypes package and I didn't hear back on that from the QueryServing team when I asked about that so it feel to the wayside.

**Comment by dbussink** [comment]
> Thank you for these! I had discussed this before with @rohit-nayak-ps but we didn't have Int32 type support in the sqltypes package and I didn't hear back on that from the QueryServing team when I asked about that so it feel to the wayside.

Ah yeah, we actually do support `int32` in a bunch of places, just not all yet. I've added this in https://github.com/vitessio/vitess/pull/12218/commits/396d4691ef965a8f7a3e9a3d1852d6af4004a362 and it's a pretty straightforward addition, so that's good :smile:.

**Review by mattlord — APPROVED** [review]
My pedantic self loves it. ❤️ The only thing I'm unsure about are the protobuf changes... they seem like potentially breaking changes, BUT I think they're all internal-only so should be fine. Could cause upgrade/downgrade issues though potentially. @deepthi thoughts?

**Comment by dbussink** [comment]
> My pedantic self loves it. ❤️ The only thing I'm unsure about are the protobuf changes... they seem like potentially breaking changes, BUT I think they're all internal-only so should be fine. Could cause upgrade/downgrade issues though potentially. @deepthi thoughts?

> Instead this now converts everything to int32. It's also safe to change the protobuf, since the protobuf wire format for int32, uint32, int64 & uint64 is all compatible and actually uses the same variable int length format.

@mattlord This is something I mentioned in the PR body, it's actually compatible :smile:. The only time it's not, is when it overflows the lower type, but as mentioned there as well, we have a different problem then :smile:.

**Comment by mattlord** [comment]
> > My pedantic self loves it. ❤️ The only thing I'm unsure about are the protobuf changes... they seem like potentially breaking changes, BUT I think they're all internal-only so should be fine. Could cause upgrade/downgrade issues though potentially. @deepthi thoughts?
> 
> > Instead this now converts everything to int32. It's also safe to change the protobuf, since the protobuf wire format for int32, uint32, int64 & uint64 is all compatible and actually uses the same variable int length format.
> 
> @mattlord This is something I mentioned in the PR body, it's actually compatible 😄. The only time it's not, is when it overflows the lower type, but as mentioned there as well, we have a different problem then 😄.

👍 Let's merge this in v17 (main becomes 17-snapshot soon) and then we have plenty of time to catch anything unexpected.

File: go/vt/vttablet/tabletmanager/vreplication/engine.go [hunk 1]
```diff
@@ -671,11 +676,15 @@ func (vre *Engine) transitionJournal(je *journalEvent) {
 			return
 		}
 		log.Infof("Created stream: %v for %v", qr.InsertID, sgtid)
-		newids = append(newids, int(qr.InsertID))
+		if qr.InsertID > math.MaxInt32 {
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletmanager/vreplication/engine.go:679]: nice

**Review by shlomi-noach — APPROVED** [review]
This is great and I like how a lot of casting is now unnecessary.

Merely wondering whether we should aim for `int64`? Not expecting workflow sequences to run out of `int32` but then again, is there a particular reason not to use `int64`?

**Comment by dbussink** [comment]
> Merely wondering whether we should aim for int64? Not expecting workflow sequences to run out of int32 but then again, is there a particular reason not to use int64?

Because the underlying table uses an `int` in MySQL, which is 32 bits. If we want to go to `int64`, we also should change the vreplication table to be a `begint`.

**Comment by shlomi-noach** [comment]
>  we also should change the vreplication table to be a bigint.

@dbussink right! And I'm happy to do that. We can do it all in one PR. If you prefer to first merge this with `int32` it's fine and we can increase later.

**Comment by shlomi-noach** [comment]
We stick to `int32` in the meantime.

**Pull request merged**